src/java.base/share/classes/java/security/spec/PKCS8EncodedKeySpec.java

```diff
@@ -63,6 +63,14 @@ public class PKCS8EncodedKeySpec extends EncodedKeySpec {
 
     /**
      * Creates a new {@code PKCS8EncodedKeySpec} with the given encoded key.
+     * <p>
+     * This constructor parses the encoded key and recovers the algorithm
+     * name from its `privateKeyAlgorithm` field. If the name is one of the
+     * <a href="{@docRoot}/../specs/security/standard-names.html#keyfactory-algorithms">
+     * standard names listed in the Java Security Standard Algorithm Names Specification</a>,
+     * it will be returned. Otherwise, the object identifier inside the `privateKeyAlgorithm`
+     * field is returned in its string format (For example, "1.3.14.7.2.1.1").
+     * If the encoded key cannot be parsed correctly, the algorithm will be null.
```

Comment:
Is this that OID format last week we talked about with passing algorithms we don't know to another provider?